src/Images/ColorRangeInput.jsx

```diff
@@ -52,19 +106,48 @@ function ColorRangeInput(props) {
   }
 
   const rangeChanged = (minVal, maxVal) => {
-    const bounds = actorContext.colorRangeBounds.get(
-      actorContext.selectedComponent
-    )
-    const rangeMin = minVal < bounds[0] ? bounds[0] : minVal
-    const rangeMax = maxVal > bounds[1] ? bounds[1] : maxVal
-    send({
-      type: 'IMAGE_COLOR_RANGE_CHANGED',
-      data: {
-        name,
-        component: actorContext.selectedComponent,
-        range: [rangeMin, rangeMax]
+    const bounds = boundsSelected
+
+    if (!isNaN(minVal) && !isNaN(maxVal)) {
+      let rangeMax = maxVal >= bounds[1] ? bounds[1] : maxVal
+      let rangeMin = minVal <= bounds[0] ? bounds[0] : minVal
+
+      // if have maxVal and minVal not respecting signs,
+      // display the numbers but do not send them up
+      if (minVal >= maxVal || maxVal <= minVal) {
+        if (maxVal <= bounds[0]) {
+          setmaxIntent(bounds[0] + 1)
+        } else if (maxVal === maxIntent || isNaN(prevMaxVal)) {
+          setmaxIntent(maxVal)
+        } else {
+          setmaxIntent(maxIntent + maxVal - prevMaxVal)
+        }
+        if (minVal >= bounds[1]) {
+          setminIntent(bounds[1] - 1)
+        } else if (minVal === minIntent || isNaN(prevMinVal)) {
+          setminIntent(minVal)
+        } else {
+          setminIntent(minIntent + minVal - prevMinVal)
+        }
+      } else {
+        setmaxIntent(rangeMax)
+        setminIntent(rangeMin)
+        send({
+          type: 'IMAGE_COLOR_RANGE_CHANGED',
+          data: {
+            name,
+            component: selectedComponent,
+            range: [rangeMin, rangeMax]
+          }
+        })
       }
-    })
+    } else if (isNaN(minVal)) {
+      setminIntent('')
+    } else if (isNaN(maxVal)) {
+      setmaxIntent('')
+    }
```

Comment:
I could be missing something, but should we have a big `else` with 2 internal `if`s?
```
} else {
  if (isNaN(minVal)) {
      setminIntent('')
    } 

  if (isNaN(maxVal)) {
      setmaxIntent('')
    }
}
```